### PR TITLE
Check pointer after weak_ptr locking

### PR DESCRIPTION
**Pull request by teams2ua**
Check pointers after weak_ptr.lock()

### CODE REVIEW

File: core/src/debug/RotatingEncryptableSink.cpp [hunk 1]
```diff
@@ -125,6 +127,9 @@ namespace ledger {
 
         void RotatingEncryptableSink::_rotate() {
             auto resolver = _resolver.lock();
+			if (!resolver) {
```
Comment (KhalilBellakrid) [inline core/src/debug/RotatingEncryptableSink.cpp]: What happened with all your indentations ? it seems broken everywhere 😞

**Comment by phaazon** [comment]
(also, indentation is broken)

File: core/src/debug/RotatingEncryptableSink.cpp (hunk 1 above)
Comment (teams2ua) [reply core/src/debug/RotatingEncryptableSink.cpp]: Oh.... let me check

**Comment by teams2ua** [comment]
> Can you provide more hindsight about why you need this? i.e. if there’s an issue with a use-after-free, this PR band-aids the consequences, it doesn’t fix the cause.
It is better, because now we can have a stack-trace not just a crash. Will allow us identify problem earlier. Maybe it will help us point the problem that some people have with android

**Comment by phaazon** [comment]
Fair enough then. Thanks for doing this! I’ll approve after you’ve addressed El @KhalilBellakrid’s comment.

**Pull request merged**